### ISSUE TITLE
[JENKINS-3831] updates hang on download

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#14334]
<p>I've been running Hudson 1.301 on another system in stand-alone mode<br/>
quite well (meaning it runs under Winstone). I'm setting it up on<br/>
another system under Tomcat 6 linked to Apache via AJP. Viewing all of<br/>
the pages is working just fine. Now I'm trying to install some plugins<br/>
and get the SSH Slaves plugin update and I'm having problems.  I get to<br/>
the update page and the preparation task finishes, but the plugins all<br/>
sit in pending. I've even left it for an hour and nothing.</p>

<p>This happens 99% of the time with my hudson setups that are running inside<br/>
Tomcat. I've yet to have this happen on a hudson setup running stand-alone.<br/>
Typically if I select the updates to be installed again that gets one more<br/>
installed, so if I want to install 3 updates, I need to select all 3, then<br/>
select all 3 again, then select 2, then select the last one again.</p>

```
Thread Dump
Update center installer thread

"Update center installer thread" Id=1331 WAITING on
java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@​fe970f
    at sun.misc.Unsafe.park(Native Method)
    -  waiting on
java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@​fe970f
    at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
    at
java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:1978)
    at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:386)
    at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1043)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1103)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:603)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor16

"TP-Processor16" Id=34 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@​1ab78f8
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor15

"TP-Processor15" Id=33 WAITING on
org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@​92fb41
    at java.lang.Object.wait(Native Method)
    -  waiting on org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@​92fb41
    at java.lang.Object.wait(Object.java:502)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:662)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor14

"TP-Processor14" Id=32 WAITING on
org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@​1ada224
    at java.lang.Object.wait(Native Method)
    -  waiting on org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@​1ada224
    at java.lang.Object.wait(Object.java:502)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:662)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor13

"TP-Processor13" Id=31 WAITING on
org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@​700b2b
    at java.lang.Object.wait(Native Method)
    -  waiting on org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@​700b2b
    at java.lang.Object.wait(Object.java:502)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:662)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor12

"TP-Processor12" Id=30 WAITING on
org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@​1ed6cb1
    at java.lang.Object.wait(Native Method)
    -  waiting on org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@​1ed6cb1
    at java.lang.Object.wait(Object.java:502)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:662)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor11

"TP-Processor11" Id=29 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@​1a9dd25
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor10

"TP-Processor10" Id=28 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@​f2d3fd
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor9

"TP-Processor9" Id=27 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@​1c4526d
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor8

"TP-Processor8" Id=26 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@​35ea3e
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor7

"TP-Processor7" Id=25 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@​348cb8
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor6

"TP-Processor6" Id=24 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@​73382c
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor5

"TP-Processor5" Id=23 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@​9be34d
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)


Thread-8

"Thread-8" Id=21 TIMED_WAITING on java.util.LinkedHashSet@​680ae9
    at java.lang.Object.wait(Native Method)
    -  waiting on java.util.LinkedHashSet@​680ae9
    at hudson.model.ViewJob$ReloadThread.getNext(ViewJob.java:159)
    at hudson.model.ViewJob$ReloadThread.run(ViewJob.java:175)


Executor #0 for master

"Executor #0 for master" Id=20 WAITING on hudson.util.OneShotEvent@​a4579e
    at java.lang.Object.wait(Native Method)
    -  waiting on hudson.util.OneShotEvent@​a4579e
    at java.lang.Object.wait(Object.java:502)
    at hudson.util.OneShotEvent.block(OneShotEvent.java:60)
    at hudson.model.Queue.pop(Queue.java:596)
    at hudson.model.Executor.run(Executor.java:97)


Hudson UDP 33848 monitoring thread

"Hudson UDP 33848 monitoring thread" Id=19 RUNNABLE (in native)
    at sun.nio.ch.DatagramChannelImpl.receive0(Native Method)
    at sun.nio.ch.DatagramChannelImpl.receiveIntoNativeBuffer(DatagramChannelImpl.java:220)
    at sun.nio.ch.DatagramChannelImpl.receive(DatagramChannelImpl.java:206)
    at sun.nio.ch.DatagramChannelImpl.receive(DatagramChannelImpl.java:150)
    -  locked java.lang.Object@​1744d94
    at hudson.UDPBroadcastThread.run(UDPBroadcastThread.java:62)


TCP slave agent listener port=0

"TCP slave agent listener port=0" Id=18 RUNNABLE (in native)
    at java.net.PlainSocketImpl.socketAccept(Native Method)
    at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:358)
    at java.net.ServerSocket.implAccept(ServerSocket.java:470)
    at java.net.ServerSocket.accept(ServerSocket.java:438)
    at hudson.TcpSlaveAgentListener.run(TcpSlaveAgentListener.java:105)


TP-Monitor

"TP-Monitor" Id=16 TIMED_WAITING on
org.apache.tomcat.util.threads.ThreadPool$MonitorRunnable@​915432
    at java.lang.Object.wait(Native Method)
    -  waiting on org.apache.tomcat.util.threads.ThreadPool$MonitorRunnable@​915432
    at org.apache.tomcat.util.threads.ThreadPool$MonitorRunnable.run(ThreadPool.java:565)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor4

"TP-Processor4" Id=15 RUNNABLE (in native)
    at java.net.PlainSocketImpl.socketAccept(Native Method)
    at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:358)
    at java.net.ServerSocket.implAccept(ServerSocket.java:470)
    at java.net.ServerSocket.accept(ServerSocket.java:438)
    at org.apache.jk.common.ChannelSocket.accept(ChannelSocket.java:306)
    at org.apache.jk.common.ChannelSocket.acceptConnections(ChannelSocket.java:660)
    at org.apache.jk.common.ChannelSocket$SocketAcceptor.runIt(ChannelSocket.java:870)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor3

"TP-Processor3" Id=14 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@​7ef567
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor2

"TP-Processor2" Id=13 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@​1e9bb55
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)


TP-Processor1

"TP-Processor1" Id=12 RUNNABLE
    at sun.management.ThreadImpl.dumpThreads0(Native Method)
    at sun.management.ThreadImpl.getThreadInfo(ThreadImpl.java:377)
    at hudson.Functions.getThreadInfos(Functions.java:716)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:616)
    at org.apache.commons.jexl.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:258)
    at org.apache.commons.jexl.parser.ASTMethod.execute(ASTMethod.java:104)
    at org.apache.commons.jexl.parser.ASTReference.execute(ASTReference.java:83)
    at org.apache.commons.jexl.parser.ASTReference.value(ASTReference.java:57)
    at org.apache.commons.jexl.parser.ASTReferenceExpression.value(ASTReferenceExpression.java:51)
    at org.apache.commons.jexl.ExpressionImpl.evaluate(ExpressionImpl.java:80)
    at hudson.ExpressionFactory2$JexlExpression.evaluate(ExpressionFactory2.java:71)
    at org.apache.commons.jelly.expression.ExpressionSupport.evaluateRecurse(ExpressionSupport.java:61)
    at org.apache.commons.jelly.expression.ExpressionSupport.evaluateAsIterator(ExpressionSupport.java:94)
    at org.apache.commons.jelly.tags.core.ForEachTag.doTag(ForEachTag.java:89)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.tags.core.WhenTag.doTag(WhenTag.java:46)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.tags.core.ChooseTag.doTag(ChooseTag.java:38)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.kohsuke.stapler.jelly.CustomTagLibrary$StaplerDynamicTag$1.run(CustomTagLibrary.java:147)
    at org.apache.commons.jelly.tags.define.InvokeBodyTag.doTag(InvokeBodyTag.java:91)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.tags.core.IfTag.doTag(IfTag.java:42)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.tags.core.JellyTag.doTag(JellyTag.java:45)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.DynamicTag.doTag(DynamicTag.java:81)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.kohsuke.stapler.jelly.CustomTagLibrary$StaplerDynamicTag$1.run(CustomTagLibrary.java:147)
    at org.apache.commons.jelly.tags.define.InvokeBodyTag.doTag(InvokeBodyTag.java:91)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.impl.StaticTag.doTag(StaticTag.java:65)
    at org.apache.commons.jelly.impl.StaticTagScript.run(StaticTagScript.java:112)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.impl.StaticTag.doTag(StaticTag.java:65)
    at org.apache.commons.jelly.impl.StaticTagScript.run(StaticTagScript.java:112)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.impl.StaticTag.doTag(StaticTag.java:65)
    at org.apache.commons.jelly.impl.StaticTagScript.run(StaticTagScript.java:112)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.impl.StaticTag.doTag(StaticTag.java:65)
    at org.apache.commons.jelly.impl.StaticTagScript.run(StaticTagScript.java:112)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.impl.StaticTag.doTag(StaticTag.java:65)
    at org.apache.commons.jelly.impl.StaticTagScript.run(StaticTagScript.java:112)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.tags.core.JellyTag.doTag(JellyTag.java:45)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.DynamicTag.doTag(DynamicTag.java:81)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.tags.core.JellyTag.doTag(JellyTag.java:45)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.kohsuke.stapler.jelly.DefaultScriptInvoker.invokeScript(DefaultScriptInvoker.java:29)
    at org.kohsuke.stapler.jelly.JellyFacet$1.dispatch(JellyFacet.java:51)
    at org.kohsuke.stapler.Stapler.invoke(Stapler.java:486)
    at org.kohsuke.stapler.Stapler.invoke(Stapler.java:404)
    at org.kohsuke.stapler.Stapler.service(Stapler.java:117)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:290)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
    at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:91)
    at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:83)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
    at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)
    at hudson.security.UnwrapSecurityExceptionFilter.doFilter(UnwrapSecurityExceptionFilter.java:51)
    at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
    at org.acegisecurity.ui.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:166)
    at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
    at org.acegisecurity.providers.anonymous.AnonymousProcessingFilter.doFilter(AnonymousProcessingFilter.java:125)
    at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
    at hudson.security.BasicAuthenticationFilter.doFilter(BasicAuthenticationFilter.java:112)
    at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
    at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:76)
    at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:155)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
    at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:233)
    at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:191)
    at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:433)
    at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:128)
    at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:102)
    at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:109)
    at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:286)
    at org.apache.jk.server.JkCoyoteHandler.invoke(JkCoyoteHandler.java:190)
    at org.apache.jk.common.HandlerRequest.invoke(HandlerRequest.java:283)
    at org.apache.jk.common.ChannelSocket.invoke(ChannelSocket.java:767)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:697)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)


ContainerBackgroundProcessor[StandardEngine[Catalina]]

"ContainerBackgroundProcessor[StandardEngine[Catalina]]" Id=11 TIMED_WAITING
    at java.lang.Thread.sleep(Native Method)
    at org.apache.catalina.core.ContainerBase$ContainerBackgroundProcessor.run(ContainerBase.java:1579)
    at java.lang.Thread.run(Thread.java:636)


Hudson cron thread

"Hudson cron thread" Id=10 TIMED_WAITING on java.util.TaskQueue@​9e82d
    at java.lang.Object.wait(Native Method)
    -  waiting on java.util.TaskQueue@​9e82d
    at java.util.TimerThread.mainLoop(Timer.java:531)
    at java.util.TimerThread.run(Timer.java:484)


Java2D Disposer

"Java2D Disposer" Id=8 WAITING on java.lang.ref.ReferenceQueue$Lock@​6a1acb
    at java.lang.Object.wait(Native Method)
    -  waiting on java.lang.ref.ReferenceQueue$Lock@​6a1acb
    at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:133)
    at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:149)
    at sun.java2d.Disposer.run(Disposer.java:143)
    at java.lang.Thread.run(Thread.java:636)


Signal Dispatcher

"Signal Dispatcher" Id=4 RUNNABLE


Finalizer

"Finalizer" Id=3 WAITING on java.lang.ref.ReferenceQueue$Lock@​52c9e
    at java.lang.Object.wait(Native Method)
    -  waiting on java.lang.ref.ReferenceQueue$Lock@​52c9e
    at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:133)
    at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:149)
    at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:177)


Reference Handler

"Reference Handler" Id=2 WAITING on java.lang.ref.Reference$Lock@​1b45f09
    at java.lang.Object.wait(Native Method)
    -  waiting on java.lang.ref.Reference$Lock@​1b45f09
    at java.lang.Object.wait(Object.java:502)
    at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:133)


main

"main" Id=1 RUNNABLE (in native)
    at java.net.PlainSocketImpl.socketAccept(Native Method)
    at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:358)
    at java.net.ServerSocket.implAccept(ServerSocket.java:470)
    at java.net.ServerSocket.accept(ServerSocket.java:438)
    at org.apache.catalina.core.StandardServer.await(StandardServer.java:389)
    at org.apache.catalina.startup.Catalina.await(Catalina.java:642)
    at org.apache.catalina.startup.Catalina.start(Catalina.java:602)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:616)
    at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:288)
    at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:413)


```

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jpschewe.jpg" title="jpschewe's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jpschewe">jpschewe</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3831" target="_blank">updates hang on download</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Critical
<li><b>component(s)</b>: core
<li><b>resolution</b>: Cannot Reproduce
<li><b>resolved</b>: 2014-05-18T13:26:00+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I've been running Hudson 1.301 on another system in stand-alone mode
quite well (meaning it runs under Winstone). I'm setting it up on
another system under Tomcat 6 linked to Apache via AJP. Viewing all of
the pages is working just fine. Now I'm trying to install some plugins
and get the SSH Slaves plugin update and I'm having problems.  I get to
the update page and the preparation task finishes, but the plugins all
sit in pending. I've even left it for an hour and nothing.</p>

<p>This happens 99% of the time with my hudson setups that are running inside
Tomcat. I've yet to have this happen on a hudson setup running stand-alone.
Typically if I select the updates to be installed again that gets one more
installed, so if I want to install 3 updates, I need to select all 3, then
select all 3 again, then select 2, then select the last one again.</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Thread Dump
Update center installer thread

"Update center installer thread" Id=1331 WAITING on
java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@fe970f
    at sun.misc.Unsafe.park(Native Method)
    -  waiting on
java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@fe970f
    at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
    at
java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:1978)
    at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:386)
    at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1043)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1103)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:603)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor16

"TP-Processor16" Id=34 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@1ab78f8
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor15

"TP-Processor15" Id=33 WAITING on
org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@92fb41
    at java.lang.Object.wait(Native Method)
    -  waiting on org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@92fb41
    at java.lang.Object.wait(Object.java:502)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:662)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor14

"TP-Processor14" Id=32 WAITING on
org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@1ada224
    at java.lang.Object.wait(Native Method)
    -  waiting on org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@1ada224
    at java.lang.Object.wait(Object.java:502)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:662)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor13

"TP-Processor13" Id=31 WAITING on
org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@700b2b
    at java.lang.Object.wait(Native Method)
    -  waiting on org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@700b2b
    at java.lang.Object.wait(Object.java:502)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:662)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor12

"TP-Processor12" Id=30 WAITING on
org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@1ed6cb1
    at java.lang.Object.wait(Native Method)
    -  waiting on org.apache.tomcat.util.threads.ThreadPool$ControlRunnable@1ed6cb1
    at java.lang.Object.wait(Object.java:502)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:662)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor11

"TP-Processor11" Id=29 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@1a9dd25
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor10

"TP-Processor10" Id=28 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@f2d3fd
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor9

"TP-Processor9" Id=27 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@1c4526d
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor8

"TP-Processor8" Id=26 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@35ea3e
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor7

"TP-Processor7" Id=25 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@348cb8
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor6

"TP-Processor6" Id=24 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@73382c
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor5

"TP-Processor5" Id=23 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@9be34d
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)

Thread-8

"Thread-8" Id=21 TIMED_WAITING on java.util.LinkedHashSet@680ae9
    at java.lang.Object.wait(Native Method)
    -  waiting on java.util.LinkedHashSet@680ae9
    at hudson.model.ViewJob$ReloadThread.getNext(ViewJob.java:159)
    at hudson.model.ViewJob$ReloadThread.run(ViewJob.java:175)

Executor #0 for master

"Executor #0 for master" Id=20 WAITING on hudson.util.OneShotEvent@a4579e
    at java.lang.Object.wait(Native Method)
    -  waiting on hudson.util.OneShotEvent@a4579e
    at java.lang.Object.wait(Object.java:502)
    at hudson.util.OneShotEvent.block(OneShotEvent.java:60)
    at hudson.model.Queue.pop(Queue.java:596)
    at hudson.model.Executor.run(Executor.java:97)

Hudson UDP 33848 monitoring thread

"Hudson UDP 33848 monitoring thread" Id=19 RUNNABLE (in native)
    at sun.nio.ch.DatagramChannelImpl.receive0(Native Method)
    at sun.nio.ch.DatagramChannelImpl.receiveIntoNativeBuffer(DatagramChannelImpl.java:220)
    at sun.nio.ch.DatagramChannelImpl.receive(DatagramChannelImpl.java:206)
    at sun.nio.ch.DatagramChannelImpl.receive(DatagramChannelImpl.java:150)
    -  locked java.lang.Object@1744d94
    at hudson.UDPBroadcastThread.run(UDPBroadcastThread.java:62)

TCP slave agent listener port=0

"TCP slave agent listener port=0" Id=18 RUNNABLE (in native)
    at java.net.PlainSocketImpl.socketAccept(Native Method)
    at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:358)
    at java.net.ServerSocket.implAccept(ServerSocket.java:470)
    at java.net.ServerSocket.accept(ServerSocket.java:438)
    at hudson.TcpSlaveAgentListener.run(TcpSlaveAgentListener.java:105)

TP-Monitor

"TP-Monitor" Id=16 TIMED_WAITING on
org.apache.tomcat.util.threads.ThreadPool$MonitorRunnable@915432
    at java.lang.Object.wait(Native Method)
    -  waiting on org.apache.tomcat.util.threads.ThreadPool$MonitorRunnable@915432
    at org.apache.tomcat.util.threads.ThreadPool$MonitorRunnable.run(ThreadPool.java:565)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor4

"TP-Processor4" Id=15 RUNNABLE (in native)
    at java.net.PlainSocketImpl.socketAccept(Native Method)
    at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:358)
    at java.net.ServerSocket.implAccept(ServerSocket.java:470)
    at java.net.ServerSocket.accept(ServerSocket.java:438)
    at org.apache.jk.common.ChannelSocket.accept(ChannelSocket.java:306)
    at org.apache.jk.common.ChannelSocket.acceptConnections(ChannelSocket.java:660)
    at org.apache.jk.common.ChannelSocket$SocketAcceptor.runIt(ChannelSocket.java:870)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor3

"TP-Processor3" Id=14 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@7ef567
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor2

"TP-Processor2" Id=13 RUNNABLE (in native)
    at java.net.SocketInputStream.socketRead0(Native Method)
    at java.net.SocketInputStream.read(SocketInputStream.java:146)
    at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
    -  locked java.io.BufferedInputStream@1e9bb55
    at org.apache.jk.common.ChannelSocket.read(ChannelSocket.java:620)
    at org.apache.jk.common.ChannelSocket.receive(ChannelSocket.java:558)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:685)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)

TP-Processor1

"TP-Processor1" Id=12 RUNNABLE
    at sun.management.ThreadImpl.dumpThreads0(Native Method)
    at sun.management.ThreadImpl.getThreadInfo(ThreadImpl.java:377)
    at hudson.Functions.getThreadInfos(Functions.java:716)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:616)
    at org.apache.commons.jexl.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:258)
    at org.apache.commons.jexl.parser.ASTMethod.execute(ASTMethod.java:104)
    at org.apache.commons.jexl.parser.ASTReference.execute(ASTReference.java:83)
    at org.apache.commons.jexl.parser.ASTReference.value(ASTReference.java:57)
    at org.apache.commons.jexl.parser.ASTReferenceExpression.value(ASTReferenceExpression.java:51)
    at org.apache.commons.jexl.ExpressionImpl.evaluate(ExpressionImpl.java:80)
    at hudson.ExpressionFactory2$JexlExpression.evaluate(ExpressionFactory2.java:71)
    at org.apache.commons.jelly.expression.ExpressionSupport.evaluateRecurse(ExpressionSupport.java:61)
    at org.apache.commons.jelly.expression.ExpressionSupport.evaluateAsIterator(ExpressionSupport.java:94)
    at org.apache.commons.jelly.tags.core.ForEachTag.doTag(ForEachTag.java:89)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.tags.core.WhenTag.doTag(WhenTag.java:46)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.tags.core.ChooseTag.doTag(ChooseTag.java:38)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.kohsuke.stapler.jelly.CustomTagLibrary$StaplerDynamicTag$1.run(CustomTagLibrary.java:147)
    at org.apache.commons.jelly.tags.define.InvokeBodyTag.doTag(InvokeBodyTag.java:91)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.tags.core.IfTag.doTag(IfTag.java:42)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.tags.core.JellyTag.doTag(JellyTag.java:45)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.DynamicTag.doTag(DynamicTag.java:81)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.kohsuke.stapler.jelly.CustomTagLibrary$StaplerDynamicTag$1.run(CustomTagLibrary.java:147)
    at org.apache.commons.jelly.tags.define.InvokeBodyTag.doTag(InvokeBodyTag.java:91)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.impl.StaticTag.doTag(StaticTag.java:65)
    at org.apache.commons.jelly.impl.StaticTagScript.run(StaticTagScript.java:112)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.impl.StaticTag.doTag(StaticTag.java:65)
    at org.apache.commons.jelly.impl.StaticTagScript.run(StaticTagScript.java:112)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.impl.StaticTag.doTag(StaticTag.java:65)
    at org.apache.commons.jelly.impl.StaticTagScript.run(StaticTagScript.java:112)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.impl.StaticTag.doTag(StaticTag.java:65)
    at org.apache.commons.jelly.impl.StaticTagScript.run(StaticTagScript.java:112)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.impl.StaticTag.doTag(StaticTag.java:65)
    at org.apache.commons.jelly.impl.StaticTagScript.run(StaticTagScript.java:112)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.tags.core.JellyTag.doTag(JellyTag.java:45)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.DynamicTag.doTag(DynamicTag.java:81)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.apache.commons.jelly.impl.ScriptBlock.run(ScriptBlock.java:95)
    at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:186)
    at org.apache.commons.jelly.tags.core.JellyTag.doTag(JellyTag.java:45)
    at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:262)
    at org.kohsuke.stapler.jelly.DefaultScriptInvoker.invokeScript(DefaultScriptInvoker.java:29)
    at org.kohsuke.stapler.jelly.JellyFacet$1.dispatch(JellyFacet.java:51)
    at org.kohsuke.stapler.Stapler.invoke(Stapler.java:486)
    at org.kohsuke.stapler.Stapler.invoke(Stapler.java:404)
    at org.kohsuke.stapler.Stapler.service(Stapler.java:117)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:290)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
    at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:91)
    at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:83)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
    at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)
    at hudson.security.UnwrapSecurityExceptionFilter.doFilter(UnwrapSecurityExceptionFilter.java:51)
    at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
    at org.acegisecurity.ui.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:166)
    at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
    at org.acegisecurity.providers.anonymous.AnonymousProcessingFilter.doFilter(AnonymousProcessingFilter.java:125)
    at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
    at hudson.security.BasicAuthenticationFilter.doFilter(BasicAuthenticationFilter.java:112)
    at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
    at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:76)
    at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:155)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
    at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:233)
    at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:191)
    at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:433)
    at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:128)
    at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:102)
    at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:109)
    at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:286)
    at org.apache.jk.server.JkCoyoteHandler.invoke(JkCoyoteHandler.java:190)
    at org.apache.jk.common.HandlerRequest.invoke(HandlerRequest.java:283)
    at org.apache.jk.common.ChannelSocket.invoke(ChannelSocket.java:767)
    at org.apache.jk.common.ChannelSocket.processConnection(ChannelSocket.java:697)
    at org.apache.jk.common.ChannelSocket$SocketConnection.runIt(ChannelSocket.java:889)
    at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:690)
    at java.lang.Thread.run(Thread.java:636)

ContainerBackgroundProcessor[StandardEngine[Catalina]]

"ContainerBackgroundProcessor[StandardEngine[Catalina]]" Id=11 TIMED_WAITING
    at java.lang.Thread.sleep(Native Method)
    at org.apache.catalina.core.ContainerBase$ContainerBackgroundProcessor.run(ContainerBase.java:1579)
    at java.lang.Thread.run(Thread.java:636)

Hudson cron thread

"Hudson cron thread" Id=10 TIMED_WAITING on java.util.TaskQueue@9e82d
    at java.lang.Object.wait(Native Method)
    -  waiting on java.util.TaskQueue@9e82d
    at java.util.TimerThread.mainLoop(Timer.java:531)
    at java.util.TimerThread.run(Timer.java:484)

Java2D Disposer

"Java2D Disposer" Id=8 WAITING on java.lang.ref.ReferenceQueue$Lock@6a1acb
    at java.lang.Object.wait(Native Method)
    -  waiting on java.lang.ref.ReferenceQueue$Lock@6a1acb
    at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:133)
    at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:149)
    at sun.java2d.Disposer.run(Disposer.java:143)
    at java.lang.Thread.run(Thread.java:636)

Signal Dispatcher

"Signal Dispatcher" Id=4 RUNNABLE

Finalizer

"Finalizer" Id=3 WAITING on java.lang.ref.ReferenceQueue$Lock@52c9e
    at java.lang.Object.wait(Native Method)
    -  waiting on java.lang.ref.ReferenceQueue$Lock@52c9e
    at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:133)
    at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:149)
    at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:177)

Reference Handler

"Reference Handler" Id=2 WAITING on java.lang.ref.Reference$Lock@1b45f09
    at java.lang.Object.wait(Native Method)
    -  waiting on java.lang.ref.Reference$Lock@1b45f09
    at java.lang.Object.wait(Object.java:502)
    at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:133)

main

"main" Id=1 RUNNABLE (in native)
    at java.net.PlainSocketImpl.socketAccept(Native Method)
    at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:358)
    at java.net.ServerSocket.implAccept(ServerSocket.java:470)
    at java.net.ServerSocket.accept(ServerSocket.java:438)
    at org.apache.catalina.core.StandardServer.await(StandardServer.java:389)
    at org.apache.catalina.startup.Catalina.await(Catalina.java:642)
    at org.apache.catalina.startup.Catalina.start(Catalina.java:602)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:616)
    at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:288)
    at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:413)

</pre>
</div></div></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Platform: All, OS: All</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>tknapstad:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3831?focusedId=130980&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-130980">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm still seeing this in 1.323. The weird thing is that once in a while one of
the plugin updates slip through and gets updated correctly. Looks like there may
be a deadlock in the synchronization somewhere:</p>

<p>Update center installer thread</p>

<p>"Update center installer thread" Id=68 WAITING on
java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@4ec4396a
	at sun.misc.Unsafe.park(Native Method)</p>
<ul class="alternate" type="square">
	<li>waiting on
java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@4ec4396a
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:158)
	at
java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:1925)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:358)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:947)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:907)
	at java.lang.Thread.run(Thread.java:619)</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>I'm still seeing this in 1.323. The weird thing is that once in a while one of<br/>
the plugin updates slip through and gets updated correctly. Looks like there may<br/>
be a deadlock in the synchronization somewhere:</p>

<p>Update center installer thread</p>

<p>"Update center installer thread" Id=68 WAITING on<br/>
java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@​4ec4396a<br/>
	at sun.misc.Unsafe.park(Native Method)</p>
<ul class="alternate" type="square">
	<li>waiting on<br/>
java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@​4ec4396a<br/>
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:158)<br/>
	at<br/>
java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:1925)<br/>
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:358)<br/>
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:947)<br/>
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:907)<br/>
	at java.lang.Thread.run(Thread.java:619)</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>tknapstad:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3831?focusedId=130981&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-130981">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I finally found a very tedious workaround for this problem which might also give
some pointers to what the problem is. </p>

<p>Say I have three available plugin updates for A,B,C. 
1) I select all those and click Install. The preparation step completes
successfully, but the plugins just say "Pending". 
2) I go back to the "Manage plugins" page, reselect all three plugins and click
install once again. I end up with six plugins in the pending list (each plugin
has a duplicate). This time however, plugin A will typically install successfully. 
3) I then go back to "Manage plugins", reselect plugins B and C which have yet
not been installed. Now I have three duplicate entries in the pending list and
plugin B installs.
4) Back to "Manage plugins" once again and select plugin C. This now installs
and I can restart Hudson with updates for all of A, B and C installed.</p>
</pre>
</details>

</ul>
</details>

<p>I finally found a very tedious workaround for this problem which might also give<br/>
some pointers to what the problem is. </p>

<p>Say I have three available plugin updates for A,B,C. <br/>
1) I select all those and click Install. The preparation step completes<br/>
successfully, but the plugins just say "Pending". <br/>
2) I go back to the "Manage plugins" page, reselect all three plugins and click<br/>
install once again. I end up with six plugins in the pending list (each plugin<br/>
has a duplicate). This time however, plugin A will typically install successfully. <br/>
3) I then go back to "Manage plugins", reselect plugins B and C which have yet<br/>
not been installed. Now I have three duplicate entries in the pending list and<br/>
plugin B installs.<br/>
4) Back to "Manage plugins" once again and select plugin C. This now installs<br/>
and I can restart Hudson with updates for all of A, B and C installed.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jpschewe.jpg" title="jpschewe's avatar" /> jpschewe:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3831?focusedId=130982&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-130982">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I've seen that behavior work as well.</p>
</pre>
</details>

</ul>
</details>

<p>I've seen that behavior work as well.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>rsutphin:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3831?focusedId=130983&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-130983">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Another confirmation of this issue.  It happens for me both on updates and on
new plugin installs in Hudson 1.324.  (It also seemed to happen on the previous
version I had installed &#8211; 1.294, I believe.)  </p>

<p>I am also running Hudson in Tomcat 6 (specifically, 6.0.18) behind Apache on
Linux, though I use mod_proxy with HTTP (not AJP).</p>

<p>$ java -version
java version "1.5.0_14"
Java(TM) 2 Runtime Environment, Standard Edition (build 1.5.0_14-b03)
Java HotSpot(TM) 64-Bit Server VM (build 1.5.0_14-b03, mixed mode)</p>

<p>The workaround described upthread (repeatedly queuing the plugins to be
installed) works sometimes for updates but not all the time (in keeping with
this being a threading issue).  I've never seen it work for new installs.</p>

<p>Manually installing plugins works fine.</p>
</pre>
</details>

</ul>
</details>

<p>Another confirmation of this issue.  It happens for me both on updates and on<br/>
new plugin installs in Hudson 1.324.  (It also seemed to happen on the previous<br/>
version I had installed &#8211; 1.294, I believe.)  </p>

<p>I am also running Hudson in Tomcat 6 (specifically, 6.0.18) behind Apache on<br/>
Linux, though I use mod_proxy with HTTP (not AJP).</p>

<p>$ java -version<br/>
java version "1.5.0_14"<br/>
Java(TM) 2 Runtime Environment, Standard Edition (build 1.5.0_14-b03)<br/>
Java HotSpot(TM) 64-Bit Server VM (build 1.5.0_14-b03, mixed mode)</p>

<p>The workaround described upthread (repeatedly queuing the plugins to be<br/>
installed) works sometimes for updates but not all the time (in keeping with<br/>
this being a threading issue).  I've never seen it work for new installs.</p>

<p>Manually installing plugins works fine.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> mindless:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3831?focusedId=139182&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-139182">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Do you still see this issue on a recent Hudson release?</p>
</pre>
</details>

</ul>
</details>

<p>Do you still see this issue on a recent Hudson release?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jpschewe.jpg" title="jpschewe's avatar" /> jpschewe:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3831?focusedId=139585&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-139585">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>My installation that was running behind Apache isn't active now, so I can't test it there. All of my other installations are now standalone. rsutphin are you still seeing this issue?</p>
</pre>
</details>

</ul>
</details>

<p>My installation that was running behind Apache isn't active now, so I can't test it there. All of my other installations are now standalone. rsutphin are you still seeing this issue?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3831?focusedId=200648&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-200648">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Multiple requests about the current status of this issue went unanswered, to resolving as Cannot Reproduce.</p>

<p>Just file a new issue if this happens again in recent versions of Jenkins.</p>
</pre>
</details>

</ul>
</details>

<p>Multiple requests about the current status of this issue went unanswered, to resolving as Cannot Reproduce.</p>

<p>Just file a new issue if this happens again in recent versions of Jenkins.</p>
